### PR TITLE
[circle-mpqsolver] Use PatternSolver

### PR DESCRIPTION
**Pull request by stamalakhov**
This commit introduces --patterns option for circle-mpqsolver and implements --LayerNorm option using PatternSolver.

Its correctness is tested at https://github.com/Samsung/ONE/pull/11560.

Draft: https://github.com/Samsung/ONE/pull/11560
Related: https://github.com/Samsung/ONE/issues/11543

ONE-DCO-1.0-Signed-off-by: s.malakhov <s.malakhov@partner.samsung.com>

### CODE REVIEW

**Comment by seanshpark** [comment]
@nnfw-bot test nncc-release

File: compiler/circle-mpqsolver/src/CircleMPQSolver.cpp [hunk 1]
```diff
@@ -47,6 +48,8 @@ int handleAutoAlgorithm(arser::Arser &arser, mpqsolver::bisection::BisectionSolv
 int entry(int argc, char **argv)
 {
   const std::string bisection_str = "--bisection";
+  const std::string patterns_str = "--patterns";
+  const std::string layernorm_str = "--LayerNorm";
```
Comment (jinevening) [inline compiler/circle-mpqsolver/src/CircleMPQSolver.cpp:52]: Could you update the option name?

```suggestion
  const std::string layernorm_str = "--q8_layernorm_with_q16_variance";
```

File: compiler/circle-mpqsolver/src/CircleMPQSolver.cpp [hunk 2]
```diff
@@ -65,10 +69,21 @@ int entry(int argc, char **argv)
...
 
+  arser.add_argument(patterns_str)
+    .nargs(0)
+    .required(false)
+    .help("Argument to define patterns applied (LayerNorm is the only supported) ");
+
+  arser.add_argument(layernorm_str)
+    .nargs(0)
+    .required(false)
+    .help("Whether to apply LayerNorm pattern");
```
Comment (jinevening) [inline compiler/circle-mpqsolver/src/CircleMPQSolver.cpp:85]:
```suggestion
    .help("Use int16 for variance computation in uint8 layer normalization");
```

File: compiler/circle-mpqsolver/src/CircleMPQSolver.cpp (hunk 1 above)
Comment (stamalakhov) [reply compiler/circle-mpqsolver/src/CircleMPQSolver.cpp:52]: > Could you update the option name?

Yes.

**Review by jinevening — APPROVED** [review]
LGTM

**Review by seanshpark — APPROVED** [review]
LGTM

**Pull request merged**